src/AlgAss/radical.jl

```diff
@@ -0,0 +1,438 @@
+    RR = _radical(AA)
+    return [A(map_entries(c -> preimage(f, c), matrix(a)), check = false) for a in RR]
+  else
+    # this is a Zech representation, so use the generic fallback
+    return _radical_finite_generic(A)
+  end
+end
+
+function _radical_finite_prime_field(A::MatAlgebra)
+  F = base_ring(A)
+  p = characteristic(F)
+  @assert p == order(F)
+  k = flog(ZZRingElem(dim(A)), p)
+
+  # We have to take the matrix trace : M_n(K) -> K
+  # and not the "algebra" trace
+  MF = trace_matrix(A, x -> tr(matrix(x, copy = false)))
+  d, B = nullspace(MF)
+  if d == 0
+    return elem_type(A)[]
+  end
+
+  C = transpose(B)
+  # Now, iterate: we need to find the kernel of tr((xy)^(p^i))/p^i mod p
+  # on the subspace generated by C
+  # Hard to believe, but this is linear!!!!
+  pl = 1
+  # if k > 0, then p < dim(A) <= typemax(Int)
+  if k > 0 
+    _p = Int(p)
+    a = A()
+    for l = 1:k
+      pl = pl * _p
+      R, mR = residue_ring(ZZ, pl * _p)
+      MR = zero_matrix(R, degree(A), degree(A))
+      M = zero_matrix(F, dim(A), nrows(C))
+      for i = 1:nrows(C)
+        c = elem_from_mat_row(A, C, i)
+        for j = 1:dim(A)
+          a = mul!(a, c, A[j])
+          MF = matrix(a, copy = false)
+          # this is suboptimal
+          lift!(MR, MF)
+          t = preimage(mR, tr(MR^pl))
+          @assert iszero(mod(t, pl))
+          M[j, i] = F(divexact(t, pl))
+        end
+      end
+      d, B = nullspace(M)
+      if d == 0
+        return elem_type(A)[]
+      end
+      C = transpose(B)*C
+    end
+  end
+
+  return elem_type(A)[ elem_from_mat_row(A, C, i) for i = 1:nrows(C) ]
+end
+
+function _radical_finite_prime_field(A::AbstractAssociativeAlgebra)
+  F = base_ring(A)
+  p = characteristic(F)
+  @assert p == order(F)
+  k = flog(ZZRingElem(dim(A)), p)
+
+  MF = trace_matrix(A)
+  d, B = nullspace(MF)
+  if d == 0
+    return elem_type(A)[]
+  end
+
+  C = transpose(B)
+  # Now, iterate: we need to find the kernel of tr((xy)^(p^i))/p^i mod p
+  # on the subspace generated by C
+  # Hard to believe, but this is linear!!!!
+  MZ = zero_matrix(ZZ, dim(A), dim(A))
+  pl = ZZRingElem(1)
+  a = A()
+  @show nrows(C)
+  @show C
+  for l = 1:k
+    pl = p*pl
+    M = zero_matrix(F, dim(A), nrows(C))
+    for i = 1:nrows(C)
+      c = elem_from_mat_row(A, C, i)
+      for j = 1:dim(A)
+        a = mul!(a, c, A[j])
+        @show i, j, a
+        MF = representation_matrix(a)
+        @show MF
+        for m = 1:nrows(MF)
+          for n = 1:ncols(MF)
+            MZ[m, n] = lift(ZZ, MF[m, n])
+          end
+        end
+        t = tr(MZ^Int(pl))
+        @assert iszero(mod(t, pl))
+        M[j, i] = F(divexact(t, pl))
+      end
+    end
+    d, B = nullspace(M)
+    if d == 0
+      return elem_type(A)[]
+    end
+    C = transpose(B)*C
+  end
+
+  return elem_type(A)[ elem_from_mat_row(A, C, i) for i = 1:nrows(C) ]
+end
+
+#function _radical(A::AbstractAssociativeAlgebra{T}) where { T <: Union{ fpFieldElem, FpFieldElem } }
```

Comment:
Is this necessary to keep this commented block ? It seems to be somehow similar to what is written above.